tests/scripts/check_files.py

```diff
@@ -263,6 +263,45 @@ def check_file_for_issue(self, filepath):
                 self.files_with_issues[filepath] = None
 
 
+class UnicodeIssueTracker(LineIssueTracker):
+    """Track lines with invalid characters or invalid text encoding."""
+
+    heading = "Invalid UTF-8 or forbidden character:"
```

Comment:
It would be nice to distinguish between the two and report the specific forbidden character, but it's not easy with the current structure of the script. I have a plan to handle that in a follow-up: https://github.com/Mbed-TLS/mbedtls/pull/2506 restructures the file and allows checkers to report custom error messages.